src/App.tsx

```diff
@@ -1,61 +1,127 @@
+import { useState } from 'react';
 import './App.scss';
-
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { TodoWithUser } from './types/TodoWithUser';
+import { User } from './types/User';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
 export const App = () => {
+  const getUserById = (userId:number, users:User[]) => {
+    const foundUser = users.find(user => user.id === userId);
+
+    return foundUser || null;
+  };
+
+  const todosWithUser: TodoWithUser[] = todosFromServer.map(todo => ({
+    ...todo,
+    user: getUserById(todo.userId, usersFromServer),
+  }));
+
+  const [todos, setTodos] = useState<TodoWithUser[]>(todosWithUser);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [todoUserId, settodoUserId] = useState(0);
+  const [hasUserIdError, setUserIdError] = useState(false);
+
+  const generateId = (getTodos:TodoWithUser[]) => {
+    const id = Math.max(...getTodos.map((todo) => todo.id));
+
+    return id + 1;
+  };
+
+  const addNewTodo = (todoTitle1: string, todoUserId1: number) => {
```

Comment:
better to use variables without numbers - instead of todoTitle1 you can use newTitle or something like that